### PR TITLE
Add guide for exporting non-YOLO PyTorch models

### PR DESCRIPTION
**Pull request by raimbekovm**
## Summary

- Adds a new docs guide `docs/en/guides/export-non-yolo-models.md` covering Ultralytics standalone export utilities for any `torch.nn.Module` (timm, torchvision, custom models)
- Walks through exporting to ONNX, TorchScript, OpenVINO, CoreML, TF SavedModel, TF Frozen Graph, NCNN, MNN, PaddlePaddle, and ExecuTorch with working code, dependencies, and platform constraints
- Includes a numerical-parity verification snippet (ONNX Runtime) and a Known Limitations section for multi-input/ExecuTorch/CoreML Python wheel caveats
- Wires the new guide into `docs/en/guides/index.md` and `mkdocs.yml` nav

## Verification

All claims in the guide were validated against the source in `ultralytics/utils/export/*`:

- Quick Start ResNet-18 ONNX export runs end-to-end and the verification snippet yields a max diff of ~7.6e-06 (< 1e-5)
- Default ONNX opset (14), input name (`images`), and output name (`output0`) match what `torch2onnx` writes
- OpenVINO export produces `model.xml` + `model.bin` with input name `x` for generic models
- NCNN export produces `model.ncnn.param`, `model.ncnn.bin`, and `model_ncnn.py`
- CoreML `BlobWriter not loaded` warning reproduced on Python 3.14 (confirming the Python 3.10-3.13 recommendation)
- Multi-input tuple support verified for ONNX and TorchScript

## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

### 🌟 Summary
📘 This PR adds a new Ultralytics guide showing how to export **non-YOLO PyTorch models**—including timm, torchvision, and custom models—into many deployment formats using one consistent export workflow.

### 📊 Key Changes
- Added a brand-new documentation page: **`export-non-yolo-models.md`** 🆕
- Introduced a guide for exporting any `torch.nn.Module` with Ultralytics export utilities, not just YOLO models 🔄
- Documented support for multiple export targets, including:
  - ONNX
  - TorchScript
  - OpenVINO
  - CoreML
  - TensorFlow SavedModel
  - TensorFlow Frozen Graph
  - NCNN
  - MNN
  - PaddlePaddle
  - ExecuTorch 🚀
- Included step-by-step code examples using a standard PyTorch/timm model to make the process easier to follow 🧩
- Added guidance on:
  - required dependencies
  - output file structures
  - dynamic shapes
  - FP16 and INT8 quantization where supported
  - platform limitations and compatibility notes ⚙️
- Added validation advice for checking exported model accuracy against the original PyTorch model ✅
- Linked the new guide into the docs navigation in both:
  - `docs/en/guides/index.md`
  - `mkdocs.yml` 🔗

### 🎯 Purpose & Impact
- Makes Ultralytics export tools more accessible to users working with **general PyTorch models**, not only YOLO models 🌍
- Reduces deployment complexity by showing a **single, unified export interface** across many formats 🛠️
- Helps developers move models to production faster on edge, mobile, and cross-platform runtimes 📦
- Clarifies supported workflows, dependencies, and limitations upfront, which can reduce trial-and-error and setup friction 💡
- Broadens the practical value of Ultralytics tooling for the wider PyTorch ecosystem, potentially attracting users with non-YOLO use cases 📈

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @raimbekovm, thank you for submitting a `ultralytics/ultralytics` 🚀 PR! This is an automated message to help with review readiness—an engineer will assist you soon. Since this PR adds a new docs guide for non-YOLO model exports, please review the checklist below to help ensure a smooth integration. ✨

-✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
-✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
-✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
-✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
-✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
-✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
-✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_  — Bruce Lee

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don't hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Good addition overall—the guide is comprehensive and well-structured. The main issues are both structural: the new links in `docs/en/guides/index.md` and `mkdocs.yml` appear mis-indented relative to neighboring entries, which can break rendering/navigation.

💬 Posted 2 inline comments

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review 2

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Useful guide overall and the coverage is strong, but there are two reproducibility gaps in the examples and one `mkdocs.yml` indentation issue that should be fixed before merge.

💬 Posted 3 inline comments

File: mkdocs.yml [hunk 1]
```diff
@@ -526,6 +526,7 @@ nav:
       - Docker Quickstart: guides/docker-quickstart.md
       - Edge TPU on Raspberry Pi: guides/coral-edge-tpu-on-raspberry-pi.md
       - End-to-End Detection: guides/end2end-detection.md
+      - Export Non-YOLO Models: guides/export-non-yolo-models.md
```
Comment (UltralyticsAssistant) [inline mkdocs.yml:529]: ⚠️ **HIGH**: This nav entry is indented less than the surrounding items in the same list. In `mkdocs.yml`, that changes the YAML nesting level, so the page can end up outside the intended Guides section or break nav parsing altogether. Please align this line's indentation with the adjacent guide entries.

File: docs/en/guides/export-non-yolo-models.md [hunk 2]
```diff
@@ -0,0 +1,275 @@
...
+The function returns a Keras model and also generates TFLite files (`.tflite`) inside `resnet18_saved_model/`.
+
+Requirements:
+
+- `tensorflow>=2.0.0,<=2.19.0`
+- `onnx2tf>=1.26.3,<1.29.0`
+- `tf_keras<=2.19.0`
+- `sng4onnx>=1.0.1`
+- `onnx_graphsurgeon>=0.3.26` (install with `--extra-index-url https://pypi.ngc.nvidia.com`)
+- `ai-edge-litert>=1.2.0` (`,<1.4.0` on macOS)
```
Comment (UltralyticsAssistant) [inline docs/en/guides/export-non-yolo-models.md]: 💡 **MEDIUM**: The macOS-specific constraint is documented as a bare `,<1.4.0`, which isn't a valid requirement users can copy/paste. Spell out the full package specifier here so the dependency line is unambiguous.

**Suggested change:**
```suggestion
- `ai-edge-litert>=1.2.0,<1.4.0` on macOS (`ai-edge-litert>=1.2.0` on other platforms)
```

**Review by UltralyticsAssistant — COMMENTED** [review]
## 🔍 PR Review 3

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)</sub>

Useful guide overall and the examples are well structured, but there are two issues worth fixing before merge: the new MkDocs nav entry appears to be misindented, and one TensorFlow dependency line is ambiguous enough to mislead users during installation.

💬 Posted 2 inline comments

**Comment by glenn-jocher** [comment]
I did a cleanup pass on this guide and pushed a docs follow-up to the branch. The main fixes were removing a few overstated dependency/version claims and clarifying which export paths go through ONNX first. Pending the current CI rerun, I do not see another technical blocker from my side; this looks ready for maintainer review/merge.

**Comment by Laughing-q** [comment]
@raimbekovm looks good! I made some updates, also update the verify example from native onnxruntime to our wrapper ONNXBackend class.

**Comment by Laughing-q** [comment]
@raimbekovm we should also have docs on how to use different backends from ultralytics to run other models as well, can you do it in a separate pr?

**Review by Laughing-q — APPROVED** [review]

**Comment by UltralyticsAssistant** [comment]
🎉 Huge congratulations on the merge of this excellent PR, @raimbekovm — and big thanks as well to @glenn-jocher and @Laughing-q for helping bring it across the finish line.

This new guide is a fantastic step forward for Ultralytics. By making the export workflow accessible for general PyTorch models, not just YOLO models, you’ve opened the door for more developers to deploy with confidence across a wide range of runtimes and platforms.

> “The best way to predict the future is to create it.” — Peter Drucker

That feels especially fitting here: this PR doesn’t just document a feature, it helps shape a more open and practical future for model deployment across the broader PyTorch ecosystem. Thank you for the clarity, care, and real-world usefulness packed into this contribution — it will help a lot of users. 🚀
